### PR TITLE
Feature/update user

### PR DESCRIPTION
**Pull request by Rodrigo-Miranda-0**
#### Board:
* [Ticket #drA25fiE](https://trello.com/c/drA25fiE/7-3-as-a-user-i-should-be-able-to-edit-my-profile-so-i-can-update-my-personal-information-or-fill-incomplete-fields)
---
#### Description:

* Allows a user to update his personal info
---
#### Notes:
*
---
#### Tasks:
- [x] Update user functionality
---
#### Risk:
*
---
#### Preview:

### CODE REVIEW

File: app/controllers/api/v1/user_controller.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
+module Api
+  module V1
+    class UserController < ApiController
+      def update
+        if current_user.id == params[:id].to_i
+          current_user.update(user_params)
```
Comment (jpascual1994) [inline app/controllers/api/v1/user_controller.rb]: Please always use the bang version
```suggestion
          current_user.update!(user_params)
```

File: app/controllers/api/v1/user_controller.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+          render json: current_user, status: :ok
+        else
+          render_unauthorized
+        end
+      end
+
+      private
+
+      def user_params
+        params.permit(:name, :lastname, :gender)
```
Comment (jpascual1994) [inline app/controllers/api/v1/user_controller.rb]: Please use snake_case, maybe you'll have to rename the DB column
```suggestion
        params.permit(:name, :last_name, :gender)
```

File: db/schema.rb [hunk 3]
```diff
@@ -66,11 +66,18 @@
     t.string "provider", default: "email", null: false
     t.string "uid", default: "", null: false
     t.string "encrypted_password", default: "", null: false
+    t.string "confirmation_token"
+    t.datetime "confirmed_at"
+    t.datetime "confirmation_sent_at"
+    t.string "unconfirmed_email"
```
Comment (jpascual1994) [inline db/schema.rb]: Should it be here? I think you have locally run a migration that isn't on the scope of this PR

File: spec/requests/api/v1/users/update_spec.rb [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      let(:headers) { another_usr.create_new_auth_token }
+
+      it 'should return an unauthorized status' do
+        subject
+        expect(response).to have_http_status(:unauthorized)
+      end
+
+      it 'should return the error message (Uanuthorized)' do
+        subject
+        p response.body
```
Comment (jpascual1994) [inline spec/requests/api/v1/users/update_spec.rb]: Please remove this line

File: app/controllers/api/v1/user_controller.rb (hunk 2 above)
Comment (juanmferreira93) [reply app/controllers/api/v1/user_controller.rb]: you also will need to update the model and migration related

**Review by juanmferreira93 — APPROVED** [review]
Left the approval but please address all the comment before proceed   🙏🏽.
Good job

**Comment by Rodrigo-Miranda-0** [comment]
What should i do about the `rails_best_practices` error? 
I believe the smell is meant to check whether a user owns a resource or not but im using the comparison to check if its the user trying to update itself. Ive read the suggested [documentation ](https://rails-bestpractices.com/posts/2010/07/20/use-scope-access/) but i don't see how i can apply that refactoring to my code, any suggestions?

**Review by juanmferreira93 — APPROVED** [review]

File: config/routes.rb [hunk 5]
```diff
@@ -4,7 +4,9 @@
   namespace :api do
     namespace :v1 do
       resources :topic, only: :index
+      resources :user, only: :update
       resources :targets, only: %i[index create destroy]
+      resources :user, only: :update
```
Comment (jpascual1994) [inline config/routes.rb]: It's duplicated

**Pull request merged**